### ISSUE TITLE
Firebird wrong size of varchar in/out parameters in procedures generated for DDL

### ISSUE THREAD

**Issue by Musashi11** [dbeaver/dbeaver#10872]
#### System information: 
- OS: WIN10
- DBeaver version: Version 7.3.2.202101032114

#### Connection specification:
- Firebird 3.0.5
- jaybird-full-3.0.9.jar

#### Describe the problem you're observing:
Procedure DDL generator is using wrong approach while generating DDL. This problem exists only for input and output parameters in procedures. Body of procedure is correct. DDL generator should take varchar size from RDB$CHARACTER_LENGTH instead of RDB$FIELD_LENGTH  which is directly related with charset of database.
For db charset none varchar size RDB$CHARACTER_LENGTH  = RDB$FIELD_LENGTH, but for db where charset is utf8  varchar size 
RDB$CHARACTER_LENGTH  x 4 = RDB$FIELD_LENGTH

#### Steps to reproduce, if exist:
Below steps will produce two databases with none and utf8 charset, each db will contains simple table and procedure to show how DDL generator behave. Table DDL generator behave correctly but, procedure DDL generator using RDB$FIELD_LENGTH information instead of RDB$CHARACTER_LENGTH

Create two databases one with charset none, next with utf8
script for none charset:
```
SET SQL DIALECT 3; 

CREATE DATABASE 'd:/db_charset_none.fdb' PAGE_SIZE 8192 DEFAULT CHARACTER SET NONE; 


create table test (
	col varchar(1)
);

SET AUTODDL OFF;
SET TERM ^ ;

CREATE OR ALTER PROCEDURE proc (intext VARCHAR(1))
RETURNS (outtext varchar(1))
AS 
BEGIN SUSPEND; END ^
```

script for utf8 charset
```

SET SQL DIALECT 3; 

CREATE DATABASE 'd:/utf.fdb' PAGE_SIZE 8192 DEFAULT CHARACTER SET UTF8; 

create table test (
	col varchar(1)
);

SET AUTODDL OFF;
SET TERM ^ ;

CREATE OR ALTER PROCEDURE proc (intext VARCHAR(1))
RETURNS (outtext varchar(1))
AS 
BEGIN SUSPEND; END ^
```

Connect to generated databases, and generate DDL for table and procedure
Database with none charset
Table DDL 
```
-- TEST definition

-- Drop table

-- DROP TABLE TEST;

CREATE TABLE TEST (
	COL VARCHAR(1)
);

CREATE OR ALTER PROCEDURE PROC (INTEXT VARCHAR(1))
RETURNS (
	OUTTEXT VARCHAR(1)
)
AS
BEGIN SUSPEND; END;
```
Database with utf8 charset
```
-- TEST definition

-- Drop table

-- DROP TABLE TEST;

CREATE TABLE TEST (
	COL VARCHAR(1)
);

CREATE OR ALTER PROCEDURE PROC (INTEXT VARCHAR(4))
RETURNS (
	OUTTEXT VARCHAR(4)
)
AS
BEGIN SUSPEND; END;

```
As you can see input and output parameters in procedure are multiply by 4, but in the same time table varchar size are OK.

**Comment by LonwoLonwo** [comment]
Hello @Musashi11 
Unfortunately I'm not an expert on Firebird. And I want to understand in what cases should I use field RDB$CHARACTER_LENGTH, and in what cases should I use field RDB$FIELD_LENGTH? 

Can I always use field RDB$CHARACTER_LENGTH instead RDB$FIELD_LENGTH? And if not always, then indicate more precisely - in what cases? only for bases with UTF-8 encoding? or even bases that have an encoding? I need to write the correct condition.

**Comment by LonwoLonwo** [comment]
As I can see, problem does not occur for database with ASCII charset.

**Comment by LonwoLonwo** [comment]
Fixed in 21.0.1
Will be available [here](https://dbeaver.io/) at 2021-03-22

**Comment by uslss** [comment]
verified
